model/project_parser.go

```diff
@@ -1186,7 +1194,7 @@ func evaluateBuildVariants(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluato
 				prunedTasks := []BuildVariantTaskUnit{}
 				toRemove := []string{}
 				for _, t := range r.RemoveTasks {
-					removed, err := tse.evalSelector(ParseSelector(t))
+					removed, _, err := tse.evalSelector(ParseSelector(t))
```

Comment:
I think this matrix logic is subtly different now. Since `evalSelector` won't error for unmatched tag criteria, so it could allow unmatched criteria to go unnoticed (without any warning/error).

That being said, matrices are long deprecated, so I'm not sure we really need to enforce validation that well on it, other than to keep the existing functionality working. If you want to keep it like this, I'm alright with it, just thought it was worth pointing out.

Reply:
I think I'll just leave it like this, I don't wanna pour extra time in to this since it's blocking some work and matrix's have been deprecated long before I even started 🤷